framework/kms-keyring.md

```diff
@@ -100,51 +106,40 @@ and MAY return an AWS KMS Client that can make the following API calls in the gi
 If the client supplier cannot supply a client for the requested region,
 it MUST communicate that fact to the AWS KMS keyring.
 
-### Key IDs
-
-Key IDs is a list of strings identifying AWS KMS CMKs, in ARN format.
-This list identifies the CMKs to be used for data key encryption and decryption with this keyring.
+### Key Names
 
-Each Key ID MUST be one of the following:
+A list of strings identifying the [AWS KMS customer master keys (CMKs)](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-id)
+that the AWS KMS keyring uses to encrypt and decrypt data keys.
 
-- A CMK [alias](https://docs.aws.amazon.com/kms/latest/developerguide/programming-aliases.html) (e.g. "alias/MyCryptoKey")
-- A well-formed key ARN (e.g. arn:aws:kms:us-east-1:999999999999:key/01234567-89ab-cdef-fedc-ba9876543210)
-- A well-formed alias ARN (e.g. arn:aws:kms:us-east-1:999999999999:alias/MyCryptoKey)
+The AWS KMS keyring accepts any value for a key name that is [accepted by AWS KMS as a key identifer](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-id).
+The AWS KMS keyring does not validate any value for a key name and passes this information to AWS KMS.
 
-See [AWS Documentation](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arn-syntax-kms).
-
-Note that only key IDs in the key ARN format will ever be used for decryption.
-This is because encrypted data keys constructed by the AWS KMS keyring will always store the ID of the
-CMK used to encrypt it in key ARN format, and [OnDecrypt](#ondecrypt) checks the key ID against that
+Note that only key names in the [key ARN format](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arn-syntax-kms) are used for decryption
+because encrypted data keys constructed by the AWS KMS keyring always store the ID of the CMK used to encrypt it in the key ARN format.
+[OnDecrypt](#ondecrypt) checks the key name against that
```

Comment:
This reads awkwardly when broken up like this.

```suggestion
because encrypted data keys constructed by the AWS KMS keyring always store the ID of the CMK used to encrypt it in the key ARN format
and [OnDecrypt](#ondecrypt) checks the key name against that
```